dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/threading/ThreadStart.java

```diff
@@ -0,0 +1,71 @@
+package com.dat3m.dartagnan.program.event.core.threading;
+
+import com.dat3m.dartagnan.program.event.EventUser;
+import com.dat3m.dartagnan.program.event.core.AbstractEvent;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.visitors.EventVisitor;
+
+import java.util.Map;
+import java.util.Set;
+
+public class ThreadStart extends AbstractEvent implements EventUser {
+
+    // May be NULL, if this thread is not spawned.
+    private ThreadCreate creator;
+    //TODO: It is probably better to make the ThreadCreate conditional rather than
+    // allow ThreadStart to fail spuriously.
+    private boolean mayFailSpuriously;
+
+    public ThreadStart(ThreadCreate creator) {
+        this.creator = creator;
+
+        mayFailSpuriously = (creator != null);
```

Comment:
what happens if I use the CLI option to force thread creation? If that is the case do we call `setMayFailSpuriously()`?

Reply:
Yes, `ThreadCreation` will spawn threads and set the option based on whether thread creation is forced to succeed.
At some point, I will probably get rid of this flag again and instead let `ThreadCreation` generate a non-deterministic choice between success and failure like so:
```
// compiling "int success = pthread_create(&t, NULL; worker, arg);" to
if (*) {
    ThreadCreate(worker, arg);
    store (t, TID);
    success = 1;
} else {
    // store (t, -1);
    success = 0;
}
```
Then `ThreadCreate/ThreadStart` can always be assumed to succeed if executed.